VultisigApp/VultisigApp/Services/Thorchain/ThorchainService.swift

```diff
@@ -15,16 +15,88 @@ class ThorchainService: ThorchainSwapProvider {
     
-    func fetchBalances(_ address: String) async throws -> [CosmosBalance] {
-        guard let url = URL(string: Endpoint.fetchAccountBalanceThorchainNineRealms(address: address)) else        {
+    func fetchBalances(coin: Coin) async throws -> [CosmosBalance] {
+        guard let url = URL(string: Endpoint.fetchAccountBalanceThorchainNineRealms(address: coin.address)) else        {
             return [CosmosBalance]()
         }
         let (data, _) = try await URLSession.shared.data(for: get9RRequest(url: url))
         
         let balanceResponse = try JSONDecoder().decode(CosmosBalanceResponse.self, from: data)
+        
         return balanceResponse.balances
     }
     
+    func fetchTokens(coin: Coin) async throws -> [CoinMeta] {
+        do {
+            let balances: [CosmosBalance] =  try await fetchBalances(coin: coin)
+            
+            var coinMetaList = [CoinMeta]()
+            for balance in balances {
+                
+                let info = getTokenMetadata(for: balance.denom)
+                
+                let coinMeta = CoinMeta(
+                    chain: .thorChain,
+                    ticker: info.symbol,
+                    logo: info.logo, // We will have to move this logo to another storage
+                    decimals: 8,
+                    priceProviderId: "", // we don't know the provider ID
+                    contractAddress: balance.denom,
+                    isNativeToken: false
+                )
+                coinMetaList.append(coinMeta)
+                
+            }
+            
+            return coinMetaList
+        } catch {
+            print("Error in fetchTokens: \(error)")
+            throw error
+        }
+    }
+    
+    struct TokenMetadata {
+        let chain: String
+        let ticker: String
+        let symbol: String
+        let decimals: Int
+        let logo: String
+    }
+
+    func getTokenMetadata(for denom: String) -> TokenMetadata {
+        let decimals = 8
+        var chain = ""
+        var symbol = ""
+        var ticker = ""
+        var logo = ""
+
+        if denom.contains(".") {
+            // Switch asset: thor.fuzn
+            let parts = denom.split(separator: ".")
+            if parts.count >= 2 {
+                chain = parts[0].uppercased()
+                symbol = parts[1].uppercased()
+                ticker = parts[1].lowercased()
+            }
+        } else if denom.contains("-") {
+            let parts = denom.split(separator: "-")
+            if parts.count >= 2 {
+                chain = parts[0].uppercased()
+                symbol = parts[1].uppercased()
+                ticker = parts[1].lowercased()
+            }
+        } else {
+            // Native THORChain asset (e.g., rune)
+            chain = "THOR"
+            symbol = denom.uppercased()
+            ticker = denom.lowercased()
+        }
+
+        logo = "https://gitlab.com/rujira/ui/-/raw/main/assets/icons/\(denom).png"
```

Comment:
Let's not load the logo from gitlab, we it can be removed 

Let's put those icons in our app first , and we can later host it on our server